cascade-api/test/Test/Cascade/Api/StateMachine/Command/Task.hs

```diff
@@ -0,0 +1,610 @@
+import           Validation                     ( Validation(..) )
+import           Chronos                        ( offsetDatetimeToTime )
+import           Data.Maybe                     ( fromMaybe )
+
+commands
+  :: MonadGen g
+  => GenBase g ~ Identity => MonadIO m => MonadTest m => [Command g m Model]
+commands =
+  [ createValid
+  , createInvalid
+  , findByProjectId
+  , addNotExistingId
+  , getExisting
+  , getNotExisting
+  , updateExisting
+  , updateExistingInvalid
+  , updateNotExisting
+  , deleteExisting
+  , deleteNotExisting
+  ]
+
+-- brittany-disable-next-binding
+data Create (v :: Type -> Type) = Create
+  { projectId  :: Var Project.Id v
+  , creatable  :: Task.RawCreatable
+  }
+  deriving stock (Generic, Show)
+
+instance HTraversable Create where
+  htraverse f Create {..} = Create <$> htraverse f projectId <*> pure creatable
+
+createValid
+  :: forall g m
+   . MonadGen g
+  => GenBase g ~ Identity => MonadIO m => MonadTest m => Command g m Model
+createValid =
+  let generator :: Model Symbolic -> Maybe (g (Create Symbolic))
+      generator model = case model ^. #project . #creatables . to Map.keys of
+        []         -> Nothing
+        projectIds -> Just $ do
+          projectId  <- Gen.element projectIds
+          title      <- Gen.nonEmptyText Valid
+          deadlineAt <- FormattedOffsetDatetime <$> Gen.deadline Valid
+          let creatable = Task.RawCreatable { .. }
+          pure $ Create { .. }
+
+      execute :: Create Concrete -> m Task.Id
+      execute (Create projectId creatable) = do
+        label "[Task/Create Valid]"
+
+        response <- evalIO
+          $ Cascade.Api.Projects.Tasks.create (concrete projectId) creatable
+
+        Response.Created task <-
+          (response ^. #responseBody)
+          |> matchUnion @(Response.Created Task.Readable)
+          |> evalMaybe
+
+        let id = task ^. #id
+
+        checkEqReadableRawCreatableTask task creatable
+
+        response ^. #responseStatusCode . #statusCode === 201
+
+        footnoteShow response
+
+        pure id
+
+      update :: Ord1 v => Model v -> Create v -> Var Task.Id v -> Model v
+      update model (Create projectId creatable) id =
+          model |> #task . #creatables . at projectId ?~ newByProjectId
+         where
+          byProjectId =
+            model ^. #task . #creatables . at projectId |> fromMaybe Map.empty
+          newByProjectId = byProjectId |> at id ?~ creatable
+  in  Command generator execute [Update update]
+
+createInvalid
+  :: forall g m
+   . MonadGen g
+  => GenBase g ~ Identity => MonadIO m => MonadTest m => Command g m Model
+createInvalid =
+  let
+    generator :: Model Symbolic -> Maybe (g (Create Symbolic))
+    generator model = case model ^. #project . #creatables . to Map.keys of
+      []         -> Nothing
+      projectIds -> Just $ do
+        flag <- Gen.bool_
+        let boolToValidity     = bool Invalid Valid
```

Comment:
`Validity` is an `Enum` and `Bounded` type we can use `Gen.enumBounded` to generate it.